ceph-releases/luminous/ubuntu/16.04/daemon/disk_introspection.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+DISK_FILE=$(uname -n)-disks
+DISK_FILE_PATH=/tmp/$DISK_FILE
+
+function ami_privileged {
+  if ! blkid > /dev/null || ! stat /dev/disk/ > /dev/null; then
+    log "ERROR: I don't have enough privileges, I can't discover devices on that machine."
+    log "ERROR: run me as a privileged container with the following options"
+    log "ERROR: --privileged=true -v /dev/:/dev/"
+    exit 1
+  fi
+  # NOTE (leseb): when not running with --privileged=true -v /dev/:/dev/
+  # lsblk is not able to get device mappers path and is complaining.
+  # That's why stderr is suppressed in /dev/null
+  DISCOVERED_DEVICES=$(lsblk --output NAME --noheadings --raw --scsi)
+}
+
+function get_all_disks_without_partitions {
+  for disk in $DISCOVERED_DEVICES; do
+    if [[ "$(grep -Ec "${disk}[0-9]" /proc/partitions)" == 0 ]]; then
+      echo "/dev/$disk" >> "$DISK_FILE_PATH"
+    fi
+  done
+  if [ ! -s "$DISK_FILE_PATH" ]; then
+    log "No disk detected."
+    log "Abort mission!"
+    exit 1
+  fi
+}
+
+function store_disk_list_configmaps {
+  log "Creating configmap $DISK_FILE in the 'ceph' namespace"
+  kubectl --namespace=ceph create configmap "$DISK_FILE" --from-file="$DISK_FILE_PATH"
```

Comment:
We can query for the namespace we're in using something like `MY_NAMESPACE=$(kubectl get pods $(hostname) -o template --template="{{.metadata.namespace}}")`.

However, if you leave out the `--namespace` altogether from the `kubectl create configmap` command, I believe it will create the configmap in the namespace the pod is running in.